driver/registry_sql.go

```diff
@@ -81,6 +84,14 @@ func (m *RegistrySQL) Init(ctx context.Context) error {
 			return err
 		}
 
+		// TODO: Some keys might need option to use software key manager. Ideas how to refactor this?
+		// 	Current solution ignores HSM configuration for TLS keys and always uses software key manager for persistence.
+		if m.C.HsmEnabled() {
+			m.defaultKeyManager = hsm.NewKeyManager(m.HsmContext())
```

Comment:
I'm not sure if this is the right place to initialize hsm key manager or should key manager logic be refactored out of persistence logic. Problem here is that key manager is for all keys at once, so for example there is no option to exempt TLS keys from HSM and that might be overhead for some HSM (that do not support CKM_TLS_MASTER_KEY_DERIVE mechanism? I have to admit i'm not an expert in all possibilites of HSM) Thats why current solution ignores HSM key manager for TLS keys:
https://github.com/ory/hydra/blob/5298b1c29b0c0ab9fe149cf29057f633d271d127/cmd/server/helper_cert.go#L66-L87